### PR TITLE
Refactors entries panel by moving code to separate functions

### PR DESCRIPTION
**Pull request by gosuwachu**
There shouldn't be any functional changes.

### CODE REVIEW

**Review by Byron — CHANGES_REQUESTED** [review]
Can you elaborate on the higher goal behind the refactoring?

It seems that did does cause visual changes:

### Previously

Screenshot 2023-12-08 at 13 00 17

### Now

Screenshot 2023-12-08 at 13 00 30

I also made `clippy` mandatory, so lints now have to be fixed for PRs to pass.

File: src/interactive/widgets/entries.rs [hunk 1]
```diff
@@ -104,129 +80,222 @@ impl Entries {
...
-            }
+            let bound = draw_top_right_help(area, title, buf);
+            draw_bottom_right_help(bound, buf);
         }
     }
 }
 
+fn entry_in_view(
+    selected: &Option<petgraph::stable_graph::NodeIndex>,
+    entries: &&[EntryDataBundle],
```
[the remark this answers is not in the document]
Comment (gosuwachu) [reply src/interactive/widgets/entries.rs:112]: Thanks, vscode refactoring has done this in few places, and this is one place I have missed :)

**Comment by gosuwachu** [comment]
While working on adding the ability to sort by number of entries in each directory (same as in `ncdu`). 
Screenshot 2023-12-08 at 13 08 24

I have noticed that the render method for the entries panel starts to get too big to work with. So, I want to make it more readable by moving code to separate methods.

Good idea to make clippy fail the checks. Also, good catch that it has caused visual changes - I will fix that.

File: src/interactive/widgets/entries.rs [hunk 2]
```diff
@@ -104,129 +80,222 @@ impl Entries {
...
-                );
-            }
+            let bound = draw_top_right_help(area, title, buf);
+            draw_bottom_right_help(bound, buf);
         }
     }
 }
 
+fn entry_in_view(
+    selected: &Option<petgraph::stable_graph::NodeIndex>,
```
[the remark this answers is not in the document]
Comment (gosuwachu) [reply src/interactive/widgets/entries.rs:111]: Done

File: src/interactive/widgets/entries.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gosuwachu) [reply src/interactive/widgets/entries.rs:112]: Done

**Comment by Byron** [comment]
> While working on adding the ability to sort by number of entries in each directory (same as in `ncdu`).

Ah, great! I hope implementing this can be similar to how the aggregation aggregates the size of the folder itself.

That will be a very nice feature to have, and it is certainly nice to learn a thing or two from `ncdu`.

**Comment by gosuwachu** [comment]
> Ah, great! I hope implementing this can be similar to how the aggregation aggregates the size of the folder itself.

Yes, it is done exactly the same way.

**Comment by gosuwachu** [comment]
Thanks, looks good!

**Pull request merged**